specs/ServerCertificate.md

```diff
@@ -0,0 +1,299 @@
+  // the error status is of type `COREWEBVIEW2_WEB_ERROR_STATUS_CERTIFICATE_IS_INVALID`
+  // and trusted by the host app.
+  if (e.ErrorStatus == CoreWebView2WebErrorStatus.CertificateIsInvalid &&
+                            ValidateServerCertificate(certificate))
+  {
+    e.Handled = true;
+  }
+  else
+  {
+    // Cancel the request for other TLS certificate error types or if untrusted by the host app.
+    e.Cancel = true;
+  }
+}
+
+// Function to validate the server certificate for untrusted root or self-signed certificate.
+bool ValidateServerCertificate(CoreWebView2Certificate certificate)
+{
+    // You may want to validate certificates in different ways depending on your app and scenario.
+    // One way might be the following:
+    // First, get the list of host app trusted certificates and its thumbprint.
+
+    // Then get the last chain element using `ICoreWebView2Certificate::get_PemEncodedIssuerCertificateChain`
+    // that contains the raw data of the untrusted root CA/self-signed certificate. Get the untrusted
+    // root CA/self signed certificate thumbprint from the raw certificate data and
+    // validate the thumbprint against the host app trusted certificate list.
+
+    // Finally, return true if it exists in the host app's certificate trusted list, or otherwise return false.
+    return true;
+}
+
+// This example clears the TLS decision in response to proceeding with TLS certificate errors.
+async void ClearServerCertificateErrorOverrideCache()
+{
+  bool isSuccessful = await webView.CoreWebView2.ClearServerCertificateErrorOverrideCacheAsync();
+  string message = (isSuccessful) ?
+                        "Clear server certificate error override cache succeeded." :
+                        "Clear server certificate error override cache failed.";
+  MessageBox.Show(this, "message", "Clear server certificate error override cache");
+}
+```
+
+# API Details
+
+## Win32 C++
+``` cpp
+[uuid(4B7FF0D2-8203-48B0-ACBF-ED9CFF82567A), object, pointer_default(unique)]
+interface ICoreWebView2_11 : ICoreWebView2_10 {
+  /// Add an event handler for the ServerCertificateError event.
+  /// The ServerCertificateError event is raised when the WebView2
+  /// cannot verify server's digital certificate while loading a web page.
+  ///
+  /// This event will raise for all web resources and follows the `WebResourceRequested` event.
+  ///
+  /// With this event you have several options for responding to server
+  /// certificate error requests:
+  ///
+  /// Scenario                                                   | Handled | Cancel
+  /// ---------------------------------------------------------- | ------- | ------
+  /// Ignore the warning and continue the request                | True    | False
+  /// Cancel the request                                         | n/a     | True
+  /// Display default TLS interstitial page                      | False   | False
+  ///
```

Comment:
Some aspects of the Cancel/Handled pair are strange to me (mismatched verb tenses, "ed" form on non-readonly property, Cancel feels like a Method, only using three states of a 2x2 matrix).
However I see this pattern on at least CoreWebView2ClientCertificateRequestedEventArgs and CoreWebView2DownloadStartingEventArgs so it probably applies here as well.  

For completeness, did we consider a three-state Allow/Default/Deny property instead of the Cancel/Handled pair, using or similar to CoreWebView2PermissionState enum. (as seen in CoreWebView2PermissionRequestedEventArgs ) ?